src/main/scala/com/gu/support/workers/model/CheckoutFailureReasons.scala

```diff
@@ -0,0 +1,92 @@
+package com.gu.support.workers.model
+
+object CheckoutFailureReasons {
+
+  val all = List(
+    PaymentMethodDetailsIncorrect,
+    PaymentMethodTemporarilyDeclined,
+    PaymentMethodUnacceptable,
+    PaymentProviderUnavailable,
+    PaymentRecentlyTaken,
+    Unknown
+  )
+
+  def fromString(string: String): Option[CheckoutFailureReason] = all.find(_.asString == string)
+
+  sealed trait CheckoutFailureReason {
+    def asString: String
+  }
+
+  case object PaymentMethodDetailsIncorrect extends CheckoutFailureReason {
+    override def asString: String = "payment_details_incorrect"
+  }
+
+  case object PaymentMethodTemporarilyDeclined extends CheckoutFailureReason {
+    override def asString: String = "payment_method_temporarily_declined"
+  }
+
+  case object PaymentMethodUnacceptable extends CheckoutFailureReason {
+    override def asString: String = "payment_method_unacceptable"
+  }
+
+  case object PaymentProviderUnavailable extends CheckoutFailureReason {
+    override def asString: String = "payment_provider_unavailable"
+  }
+
+  case object PaymentRecentlyTaken extends CheckoutFailureReason {
+    override def asString: String = "payment_recently_taken"
+  }
+
+  case object Unknown extends CheckoutFailureReason {
+    override def asString: String = "unknown"
+  }
+
+  // https://stripe.com/docs/declines/codes
+  def convertStripeDeclineCode(declineCode: String): Option[CheckoutFailureReason] = declineCode match {
```

Comment:
`condOpt` could be useful here - prevents wrapping each checkout failure reason in a `Some()`

Reply:
Thanks for the tip - I didn't know about condOpt 👍